src/components/SettlementButton.tsx

```diff
@@ -212,12 +246,22 @@ function SettlementButton({
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [currency, formattedAmount, iouReport, policyID, translate, shouldHidePaymentOptions, shouldShowApproveButton, shouldDisableApproveButton]);
     const selectPaymentType = (event: KYCFlowEvent, iouPaymentType: PaymentMethodType, triggerKYCFlow: TriggerKYCFlow) => {
-        if (iouPaymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY || iouPaymentType === CONST.IOU.PAYMENT_TYPE.VBBA) {
+        if (
+            (iouPaymentType === CONST.IOU.PAYMENT_TYPE.VBBA ||
+                iouPaymentType === CONST.PAYMENT_METHODS.BUSINESS_BANK_ACCOUNT ||
+                iouPaymentType === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT) &&
+            !hasBankAccount
+        ) {
             triggerKYCFlow(event, iouPaymentType);
             BankAccounts.setPersonalBankAccountContinueKYCOnSuccess(ROUTES.ENABLE_PAYMENTS);
             return;
         }
 
+        if ((iouPaymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY || iouPaymentType === CONST.PAYMENT_METHODS.DEBIT_CARD) && !hasPaymentMethod) {
+            triggerKYCFlow(event, iouPaymentType);
+            return;
+        }
```

Comment:
This doesn't make a lot of sense. Why don't you use the conditions as they were before?